### PR TITLE
Feature/10 be log in via facebook and google preparation

### PR DESCRIPTION
**Pull request by hromogen**
- Move common logic for controllers into a base class

- Create shared API entity in the `core` package

- Add dependencies needed for the ticket

### CODE REVIEW

File: packages/server/src/models/user.entity.ts [hunk 1]
```diff
@@ -13,4 +13,10 @@ export default class User {
 
   @Column()
   public password: string;
+
+  @Column()
+  public facebook_id: string;
```
Comment (dartwlad) [inline packages/server/src/models/user.entity.ts]: Consider naming fields in camel case.

File: packages/server/src/base-classes/controller.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Router } from 'express';
+
+import { serverApi } from 'core/api';
+
+import BaseControllerInterface from '../interfaces/base-controller.interface';
+
+export default abstract class BaseController
+  implements BaseControllerInterface {
```
Comment (dartwlad) [inline packages/server/src/base-classes/controller.ts:8]: Do we really need this new line?

Comment (hromogen) [reply packages/server/src/base-classes/controller.ts:8]: Well we do have a basic interface for a BaseController, don't we? If you think an abstract class would be enough I will remove the interface at all

File: packages/server/src/models/user.entity.ts (hunk 1 above)
Comment (hromogen) [reply packages/server/src/models/user.entity.ts]: I do not mind. Just thought that in DBs they usually name fields this way

File: packages/server/src/base-classes/controller.ts (hunk 2 above)
Comment (hromogen) [reply packages/server/src/base-classes/controller.ts:8]: This article http://dotnetpattern.com/typescript-abstract-class suggests that an abstract class can implement an interface. So I will probably leave it, but remove it from the classes that extend `BaseController` what do you think?

File: packages/server/src/models/user.entity.ts [hunk 3]
```diff
@@ -13,4 +13,10 @@ export default class User {
```
Comment (karolinayanko) [inline packages/server/src/models/user.entity.ts:13]: I'd suggest use firstName and lastName instead of name

Comment (hromogen) [reply packages/server/src/models/user.entity.ts:13]: Facebook returns us only `displayName`. Would that go?

Comment (karolinayanko) [reply packages/server/src/models/user.entity.ts:13]: can we split it by space?

Comment (hromogen) [reply packages/server/src/models/user.entity.ts:13]: Sorry, just have investigated a bit more and found out that all the fields can be extracted from a Facebook account. Don't you mind if the changes would be added during the actual implementation of `passport-facebook` `passport-google`?

Comment (karolinayanko) [reply packages/server/src/models/user.entity.ts:13]: If it not break anything after (migration, db changes, created user), I'm ok

**Review by karolinayanko — APPROVED** [review]
Ok, to move name field update to the next patch

File: packages/server/src/models/user.entity.ts (hunk 1 above)
Comment (dartwlad) [reply packages/server/src/models/user.entity.ts]: I think it's better for code consistency

File: packages/server/src/base-classes/controller.ts (hunk 2 above)
Comment (dartwlad) [reply packages/server/src/base-classes/controller.ts:8]: I meant let's leave it one line:
export default abstract class BaseController implements BaseControllerInterface {

Comment (hromogen) [reply packages/server/src/base-classes/controller.ts:8]: Well, that is autocorrected

**Review by dartwlad — APPROVED** [review]

**Pull request merged**